proposals/913-components-repo/README.md

```diff
@@ -0,0 +1,461 @@
+
+This catalog also creates a durable bridge between Kubeflow Pipelines and the broader Kubeflow ecosystem: it gives the
+SDK and API server a canonical source of reusable building blocks, provides SIGs with a venue to showcase
+Kubernetes-native MLOps patterns, and offers users a distribution channel that evolves independently from the core KFP
+release train. Centralizing community-supported assets in one place keeps the SDK surface area lean while encouraging
+experimentation in third-party areas that can later graduate into core.
+
+### Goals
+
+1. Move reusable components and pipelines into a dedicated GitHub repository with clear structure and governance.
+2. Provide standardized metadata, documentation, and testing requirements for every asset.
+3. Ship an installable Python package for core (community-maintained) artifacts that is versioned to match Kubeflow
+   releases.
+4. Maintain a parallel, clearly demarcated area for third-party contributions, shipped as its own Python package that
+   tracks the same release cadence as the core catalog.
+5. Automate maintenance (e.g. stale component detection, dependency validation) to keep the catalog healthy.
+6. Provide developer onboarding materials and guidance for agents generating components/pipelines.
+
+### Non-Goals
+
+1. Operate turnkey build or publishing pipelines for third-party maintainers. We supply validation tooling and guidance,
+   but external teams continue to manage their own release automation.
+2. Backport historical component versions; only actively maintained assets will be migrated.
+3. Remove the existing `components/` directory in the main KFP repo without notice. We will announce the new repository,
+   point contributors at the new onboarding documentation, and give the community a minimum of 90 days to move their
+   assets before deleting the legacy directory.
+4. Vendor specific components. Vendors are highly encouraged to use the same repository structure in their own
+   repository.
+
+## Proposal
+
+### Repository Layout
+
+Create a new repository (e.g. `kubeflow/kfp-components`) with the following top-level layout:
+
+```
+root
+├── components
+│   ├── README.md (catalog landing page)
+│   ├── __init__.py (auto-imports all components for clean `from kfp_components import training` usage)
+│   ├── training/
+│   │   ├── README.md (category index listing each component with summaries/links)
+│   │   ├── __init__.py (re-exports all components in this category)
+│   │   └── <component-name>/
+│   │       ├── __init__.py (exposes the component entrypoint for imports)
+│   │       ├── component.py
+│   │       ├── metadata.yaml
+│   │       ├── README.md
+│   │       ├── OWNERS
+│   │       ├── example_pipelines.py
+│   │       ├── tests/
+│   │       │   └── test_component.py
+│   │       └── <supporting_files>
+│   └── ... (other categories: evaluation/, data_processing/, etc.)
+├── pipelines
```

Comment:
Curious to hear more about the rationale behind sharable pipelines (as opposed to components).

Reply:
@droctothorpe I see a few reasons:

1. Nested pipelines are supported, so sometimes various components benefit from running in parallel or in a chain and then can be used as if it were a component.
2. Provides good examples for how some of these components stitch together for common use cases (e.g. converting PDFs to markdown and inserting them in a Vector database)
3. It provides quick starts for tutorials and documentation.